### DISCUSSION TITLE
Intellisense for pyx file

### DISCUSSION THREAD

**Discussion by Greg7000** [explosion/spaCy#11817] [Help: Coding & Implementations]
Anybody knows how to get intellisense for pyx in Vscode? 

For example:
`from spacy.matcher import Matcher`

Since matcher is a pyx file, vscode intellisense (Pylance) is unable to link to source with ctrl+click, so there is no highlight on Matcher so I can't quickly access doc related to Matcher in my IDE which is frustrating.

Do you know anyways around this?

**Comment by polm** [comment]
I don't use VSCode, but it may help to know that the pyx file is Cython - a search for Cython support in VSCode shows that there have been some attempts at support, like this one:

https://github.com/guyskk/language-cython-for-vscode

**Reply by Greg7000** [reply]
Thx, not sure if I tested it or not, will try it. I might accept this answer depending on the behavior of this plugin in VsCode

**Reply by Greg7000** [reply]
> Thx, not sure if I tested it or not, will try it. I might accept this answer depending on the behavior of this plugin in VsCode

Unfortunately, Intellisense still cannot resolve the dependency of **Matcher**. The best you can do is dive in `spacy.matcher` and see the link to Matcher which does not help.

`from .matcher import Matcher`


```
from .matcher import Matcher
from .phrasematcher import PhraseMatcher
from .dependencymatcher import DependencyMatcher

__all__ = ["Matcher", "PhraseMatcher", "DependencyMatcher"]
```

**Reply by Greg7000** [reply]
> > Thx, not sure if I tested it or not, will try it. I might accept this answer depending on the behavior of this plugin in VsCode
> 
> Unfortunately, Intellisense still cannot resolve the dependency of **Matcher**. The best you can do is dive in `spacy.matcher` and see the link to Matcher which does not help.
> 
> `from .matcher import Matcher`
> 
> ```
> from .matcher import Matcher
> from .phrasematcher import PhraseMatcher
> from .dependencymatcher import DependencyMatcher
> 
> __all__ = ["Matcher", "PhraseMatcher", "DependencyMatcher"]
> ```

Best I can find so now is using python help function with

```
from spacy.matcher import Matcher
help(Matcher)
```
Which results in 
```
class Matcher(builtins.object)
 |  Matcher(vocab, validate=True)
 |  Match sequences of tokens, based on pattern rules.
 |  
 |      DOCS: https://spacy.io/api/matcher
 |      USAGE: https://spacy.io/usage/rule-based-matching
 |  
 |  Methods defined here:
 |  

 |  __call__(...)
 |      Find all token sequences matching the supplied pattern.
 |      
 |      doclike (Doc or Span): The document to match over.
 |      as_spans (bool): Return Span objects with labels instead of (match_id,
 |          start, end) tuples. ....
```

I guess this is better than nothing
